vyper/codegen/module.py

```diff
@@ -168,6 +168,9 @@ def _runtime_ir(runtime_functions, all_sigs, global_ctx):
 
     runtime = ["seq"]
 
+    # function with 0 selector may get treated as the fallback fn without
+    # this check for calldatasize < 4. For all other fns this check is redundant
+    # because a similar check for calldatasize is done in the selector table.
```

Comment:
actually this is clarifying something for me, which is that for consistency, we need the calldatasize check for selectors with trailing 0s in addition to just straight 0 selector. otherwise those will get treated as asserts, not fallbacks

Reply:
the other option is just to skip the initial calldatasize check entirely, always, and just rely on the assertion in the selector table?

Reply:
Is there a practical difference between going to fallback, versus relying on the assert? If the assert fails it should result in a REVERT, which is what the fallback function goes to also, right?

Reply:
well -- that's the default fallback function, but the user can supply their own.

Reply:
thanks! submitted a go at a more in-depth explanation